### ISSUE TITLE
UserWarning: The figure layout has changed to tight

### ISSUE THREAD

**Issue by MP-MaximilianLattka** [mwaskom/seaborn#3431]
While using the current version of `seaborn` (0.12.2) within Jupyter Notebook (JupyterLab, version 4.0.3), I'm receiving a `UserWarning` while creating FacetGrid plots with `seaborn`. This warning also occurs with the latest commit (3ef2a0f9580e5a4715be878c73751454089c945d). Please note that the plot gets rendered as expected and I'm just bothered by the warning itself.

```python
# Print Python version
import platform
print(platform.python_version())
# '3.10.11'
```

```python
# Import seaborn and print version
import seaborn as sns
print(sns.__version__)
# '0.12.2'
```

```python
# Load penguins dataset
df_penguins = sns.load_dataset("penguins")
```

```python
g = sns.displot(kind="kde", data=df_penguins, x="body_mass_g", col="species")
# 'C:\Users\<user>\AppData\Local\pypoetry\Cache\virtualenvs\myproject-dRQgBYd3-py3.10\lib\site-packages\seaborn\axisgrid.py:118: UserWarning: The figure layout has changed to tight
#  self._figure.tight_layout(*args, **kwargs)'
```

I would like to refrain from suppressing warnings altogether with the `warning` package, but it would be my last resort:

```python
import warnings
warnings.filterwarnings('ignore')
```

**Comment by mwaskom** [comment]
I believe this is caused by this matplotlib issue: https://github.com/matplotlib/matplotlib/issues/26290

There's a fix in the dev version of matplotlib so you'll need to (1) continue filtering warnings (2) downgrade (3) live with it until a new matplotlib release.

**Comment by mwaskom** [comment]
PS in the future please report the matplotlib version when reporting seaborn issues. Thanks! Going to close as there's no action to take on the seaborn side.

**Comment by RohithMacharla11** [comment]
this is a matplotlib bug. It was fixed in version 3.7.3, so you can avoid it by upgrading Matplotlib.
